heareval/tasks/pipeline.py

```diff
@@ -550,12 +667,28 @@ def requires(self):
         }
 
     def run(self):
-        for audiofile in tqdm(list(self.requires()["corpus"].workdir.glob("*.wav"))):
-            # Compare the filename with the slug.
-            # Note that the slug does not have the extension of the file
-            split = self.metadata.loc[
-                self.metadata["slug"] == audiofile.stem, "split"
-            ].values[0]
+        audiofiles = set(self.requires()["corpus"].workdir.glob("*.wav"))
+        unique_filestems = set(
+            self.metadata["unique_filestem"].drop_duplicates().values
+        )
+        audiofilenames = set([Path(f).stem for f in unique_filestems])
+        # Gross, let's never do this again
+        if self.task_config["version"].split("-")[-1] == "small":
+            # Many filestems in the metadata won't be in the
+            # small corpus as audio.
+            assert audiofilenames <= unique_filestems
+        else:
+            assert audiofilenames == unique_filestems
```

Comment:
assertion should be on the `len` of the variables

Reply:
Why can't I compare sets of filestems?